### PR TITLE
🗞️ Task composability [1/N]

### PR DESCRIPTION
**Pull request by janjakubnanista**
### In this PR

- The first step towards composable hardhat tasks that allow us, amongst others, to support custom configuration 
- The functional bits that we want to allow users to modify will be isolated into subtasks that:
  - Allow users to execute custom logic after a subtask is executed
  - Allow users to modify the subtask without rewriting the whole task

### CODE REVIEW

File: packages/ua-devtools-evm-hardhat/src/tasks/oapp/wire/subtask.configure.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    configurator = configureOApp,
+    oappFactory = createOAppFactory(createConnectedContractFactory()),
+}: TaskArgs): Promise<OmniTransaction[]> => {
+    const logger = createModuleLogger(SUBTASK_LZ_OAPP_WIRE_CONFIGURE)
+
+    logger.verbose(`Running with graph:\n\n${printJson(graph)}`)
+
+    // As an additional step, even though this task is getting called
+    // from controlled and type-safe environments (for now),
+    // we pass thr graph through a builder
```
Comment (ryandgoulding) [inline packages/ua-devtools-evm-hardhat/src/tasks/oapp/wire/subtask.configure.ts:27]: nit:  `the`

File: packages/ua-devtools-evm-hardhat/src/tasks/oapp/wire/subtask.configure.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    try {
+        return await configurator(graph, oappFactory)
+    } catch (error) {
+        logger.verbose(`Encountered an error: ${error}`)
+
+        throw new Error(`An error occurred while getting the OApp configuration: ${error}`)
+    }
+}
+
+subtask(SUBTASK_LZ_OAPP_WIRE_CONFIGURE, 'Create a list of OmniTransactions that configure your OApp')
```
Comment (ryandgoulding) [inline packages/ua-devtools-evm-hardhat/src/tasks/oapp/wire/subtask.configure.ts:52]: Can do this instead:

```
subtask(SUBTASK_LZ_OAPP_WIRE_CONFIGURE, 'Create a list of OmniTransactions that configure your OApp', action)
    .addParam('graph', 'Configuration of you OApp of type OAppOmniGraph', undefined, types.any)
    .addParam('configurator', 'Configuration function of type OAppConfigurator', undefined, types.any, true)
    .addParam('oappFactory', 'SDK factory for OApp SDK of type OAppFactory', undefined, types.any, true)
```

**Review by ryandgoulding — CHANGES_REQUESTED** [review]

File: packages/ua-devtools-evm-hardhat/src/tasks/oapp/wire/subtask.configure.ts (hunk 1 above)
Comment (janjakubnanista) [reply packages/ua-devtools-evm-hardhat/src/tasks/oapp/wire/subtask.configure.ts]: is fixed thank you they keep putting the letters too close on my keyboard

File: packages/ua-devtools-evm-hardhat/src/tasks/oapp/wire/subtask.configure.ts (hunk 2 above)
Comment (janjakubnanista) [reply packages/ua-devtools-evm-hardhat/src/tasks/oapp/wire/subtask.configure.ts]: Fixed across the board

**Review by ryandgoulding — APPROVED** [review]

**Pull request merged**